pkg/virt-config/virt-config.go

```diff
@@ -40,7 +40,7 @@ const (
 	DefaultMachineType                              = "q35"
 	DefaultCPURequest                               = "100m"
 	DefaultMemoryOvercommit                         = 100
-	DefaultEmulatedMachines                         = "q35*,pc-q35*"
+	DefaultEmulatedMachines                         = "q35*,pc-q35*,pseries*"
```

Comment:
I guess we will need a switch somewhere in the code which can use different defaults for different build-architectures? Maybe via `GOARCH`? At least until we support mixed node architectures that should work.

Reply:
Making this change as the following would be acceptable?

```
--- a/pkg/virt-config/virt-config.go
+++ b/pkg/virt-config/virt-config.go
@@ -24,12 +24,24 @@ package virtconfig
 */

 import (
+       "runtime"
+
        k8sv1 "k8s.io/api/core/v1"
        "k8s.io/apimachinery/pkg/api/resource"

        cmdv1 "kubevirt.io/kubevirt/pkg/handler-launcher-com/cmd/v1"
 )

+// Set default machine type and supported emulated machines based on architecture
+func getDefaultMachinesForArch() (string, string) {
+       if runtime.GOARCH == "ppc64le" {
+               return "pseries", "pseries*"
+       }
+       return "q35", "q35*,pc-q35*"
+}
+
+var DefaultMachineType, DefaultEmulatedMachines = getDefaultMachinesForArch()
+
 const (
        ParallelOutboundMigrationsPerNodeDefault uint32 = 2
        ParallelMigrationsPerClusterDefault      uint32 = 5
@@ -37,10 +49,8 @@ const (
        MigrationAllowAutoConverge               bool   = false
        MigrationProgressTimeout                 int64  = 150
        MigrationCompletionTimeoutPerGiB         int64  = 800
-       DefaultMachineType                              = "q35"
        DefaultCPURequest                               = "100m"
        DefaultMemoryOvercommit                         = 100
-       DefaultEmulatedMachines                         = "q35*,pc-q35*"
        DefaultLessPVCSpaceToleration                   = 10
        DefaultNodeSelectors                            = ""
        DefaultNetworkInterface                         = "bridge"
```
I am not sure how it would be possible to keep the `DefaultMachineType` and `DefaultEmulatedMachines` emulated machines as `const` given they would depend on which architecture we are running in, so I turned them into var. Would this be a problem?

The initial change I did with only `DefaultEmulatedMachines` was due to it being rejected by virt-api when I attempted to create a pseries VMI with the following error:
```
$ kubectl create -f ../vmi-ephemeral.yaml
The request is invalid: spec.domain.machine.type: spec.domain.machine.type is not supported: pseries (allowed values: [q35* pc-q35*])
```
So I needed to add the pseries* to the list.

Reply:
> Making this change as the following would be acceptable?

Yes. In general a switch sounds good. Also we could still have two constants, one `DefaultAMD64MachineType = "q35*,pc-q35*"` and `efaultPPC64MachineType = "..."` to still see the defaults better.

Reply:
Maybe it might also make sense to make the target architecture configurable, so that our components would only guess by default, but via the config-map they can also be pointed to the right direction. In general I would not require that immediately but since it will probably also drastically simplify testing, I would recommend to add it.

Reply:
I modified the patch to add the constant specifics for both arch and then I do the selection based on which arch is running the same way I did before and returned the values defined by these consts. I still need to take a look at the other suggestion you made about the config-map, given my familiarity with the whole code is pretty low it didn't came to mind how to implement this recommendation.